src/components/timeline/Ruler.js

```diff
@@ -17,28 +20,42 @@ type Props = {|
   +rangeStart: Milliseconds,
   +rangeEnd: Milliseconds,
   +width: CssPixels,
+  +unit: string,
 |};
 
 export class TimelineRuler extends PureComponent<Props> {
-  _findNiceNumberGreaterOrEqualTo(uglyNumber: Milliseconds) {
-    // Special case numbers in the seconds, minutes or hour ranges.
-    if (uglyNumber > 10000 && uglyNumber <= 48 * 3600 * 1000) {
-      for (const seconds of [15, 20, 30]) {
-        const number = seconds * 1000;
-        if (uglyNumber <= number) {
-          return number;
+  _findNiceNumberGreaterOrEqualTo(unit: string, uglyNumber: Milliseconds) {
```

Comment:
Could be a follow-up: I'd rename this method to `_pickNotchLength`, rename `uglyNumber` to `minLength`, and then call `findRoundBytesValueGreaterOrEqualTo` and `findRoundMillisecondsValueGreaterOrEqualTo` inside it, which would be new functions in `utils/format-numbers.js` which can have some unit tests.